src/app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,13 @@
+class TagsController < ApplicationController
+  LIMIT = 30
+
+  def show
+    offset = params.fetch(:offset, 0).to_i
+
+    @tag = params[:id]
+    @functions = Tag.from_param(@tag).functions.limit(LIMIT).offset(offset).order(likes_count: :desc).all
```

Comment:
this should follow the same pattern we have when looking up a record by param.

so we look it up in another method called before_action and raises an exception when not found

Reply:
Should we really move it to another `before_action` method? it depends on `@tag` and `offset`. It would be redundant if we wrote them twice, one time in the `show` method and another one in the `before_action`

https://github.com/emad-elsaid/rubyfunctions/blob/afd4526ab0463d37fd422ac249fdda3bf365b4c7/src/app/controllers/functions_controller.rb#L15